src/main/java/org/owasp/esapi/logging/java/JavaLogFactory.java

```diff
@@ -79,7 +80,8 @@ public class JavaLogFactory implements LogFactory {
         boolean logApplicationName = ESAPI.securityConfiguration().getBooleanProp(LOG_APPLICATION_NAME);
         String appName = ESAPI.securityConfiguration().getStringProp(APPLICATION_NAME);
         boolean logServerIp = ESAPI.securityConfiguration().getBooleanProp(LOG_SERVER_IP);
-        JAVA_LOG_APPENDER = createLogAppender(logUserInfo, logClientInfo, logServerIp, logApplicationName, appName);
+        boolean omitEventTypeInLogs = ESAPI.securityConfiguration().getBooleanProp(OMIT_EVENT_TYPE_IN_LOGS);
```

Comment:
For reasons noted in a comment in the LogPrefixAppender CTOR that you modifed, there is a (probably good) possibility that this property will not be found at all in the **ESAPI.properties** file, so you need to be prepared to catch `ConfigurationException` here and set `omitEventTypeInlogs` to `false` should that happen. Because we can't let a `ConfigurationException` propagate back to the caller if **Logger.OmitEventTypeInLogs** is not found. That would result in breaking people's code and as a result is just not acceptable.

Reply:
:(  Catching a runtime exception to avoid breaking feels dirty to me, but I do agree it has the unfavorable result of breaking implementations when new properties are required.

It is worth noting that there are actually 2 cases that the Exception may occur:
1.  the value is not set
2. the value is not of the expected type (cannot be converted to a boolean)

In the second case, catching this exception may result in hiding a problem with a property value.  I do think it is a less likely scenario, but still worth noting.

```
 @Override
    public Boolean getBooleanProp(String propertyName) throws ConfigurationException {
        try {
            return esapiPropertyManager.getBooleanProp(propertyName);
        } catch (ConfigurationException ex) {
            String property = properties.getProperty( propertyName );
            if ( property == null ) {
                throw new ConfigurationException( "SecurityConfiguration for " + propertyName + " not found in ESAPI.properties");
            }
            if ( property.equalsIgnoreCase("true") || property.equalsIgnoreCase("yes" ) ) {
                return true;
            }
            if ( property.equalsIgnoreCase("false") || property.equalsIgnoreCase( "no" ) ) {
                return false;
            }
            throw new ConfigurationException( "SecurityConfiguration for " + propertyName + " has incorrect " +
                    "type");
        }
    }
```

Reply:
@jeremiahjstacey - True, but the more common error is just going to be that people don't update their ESAPI.properties file regularly, so the property would be missing and I don't want ESAPI to start failing (and thus breaking code) because of that. We could have the `ConfigurationException` logged via `logSpecial()`, but I'd prefer dirty over adding a new CTOR and all the other ugliness that brings into ESAPI code.

Reply:
We can incorporate the check of property value as you mentitoned and ignore it if the property is not present. What do you think?

`if (property != null) {
                if (property.equalsIgnoreCase("true") || property.equalsIgnoreCase("yes")) {
                    omitEventTypeInLogs = true;
                } else if (property.equalsIgnoreCase("false") || property.equalsIgnoreCase("no")) {
                    omitEventTypeInLogs = false;
                } else if (property != null) {
                    throw new ConfigurationException("SecurityConfiguration for " + OMIT_EVENT_TYPE_IN_LOGS + " has incorrect " +
                            "type");
                }
            }`

Reply:
The code I posted is the default implementation of getBooleanProp from the DefaultSecurityProvider.  I just wanted to identify the alternative case that is being swallowed.

It is known and is still preferred to the probability of breaking an environment due to a missing property.

@RodolfoAndre  You do not need to make any additional changes for this  as far as I am concerned.